nixos/modules/services/hardware/keyd.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta = {
+    maintainers = [ "cidkid" ];
+  };
+
+  options = {
```

Comment:
Probably worth adding options `applicationSpecificMapping.enable` and `applicationSpecificMapping.config` if you are building it with Python (which is not required for the base version). Also, I don't see `keyd-application-mapper` being called anywhere in the systemd service.

Reply:
Keyd recommends to use your wm or an autostart to launch keyd-application-mapper, and we can't define it in a declarative way due to the config path being in the users $HOME

Reply:
You could write a systemd user service (i.e. `systemd.user.services.keyd-application-mapper`)

Reply:
There are users created here and now talks of running it as a user service. If it can run as a user service, it definitely doesn't need a hardcoded user.